Update "how to create a new genesis block" - Closes 1526 (#1668)
LiskHQ/lisk-docs: docs/modules/ROOT/pages/build-blockchain/create-genesis-block.adoc

----
├── bin/
├── config/
│    └── default/
│         ├── config.json <1>
│         ├── dev-validators.json <2>
│         ├── genesis_assets.json <3>
│         ├── genesis_block.blob <4>
│         └── passphrase.json <5>
├── src/
└── test/
----

<1> The default configuration of the sidechain client exists in the `config.json` file.
<2> `dev-validators.json`: Contains the plain & encrypted keys, addresses, and other properties of 102 validators.
<3> `genesis_assets.json`: Contains details about the initial state of a sidechain such as account details for `token` and `pos` modules. It also contains the list of validators, etc.
<4> `genesis_block.blob`: A blob file of the genesis block.
<5> `passphrase.json`: Contains the passphrase used in various sidechain client operations.

== Why create a new genesis block?
For every new sidechain, a unique genesis block is required. 
As mentioned earlier, a default genesis block becomes available right after the client's `init` process. 
However, while running a public network, the developers have to create a new genesis block to change the initial state.
The initial genesis block is expected to be used only for the development purposes.

> To achieve this you can regenerate a genesis block for various use cases, such as assigning different tokens to specific addresses of your choice. Furthermore, it is also possible to recreate the genesis block based on the module's requirements, for example, setting up the initial constraints required for a staking module, etc.

Author: ```suggestion
To achieve this you can regenerate a genesis block for various use cases, such as assigning different tokens to specific addresses of your choice. 
Furthermore, it is also possible to recreate the genesis block based on the module's requirements, for example, setting up the initial constraints required for a staking module, etc.
```

.Default genesis block and configuration files
----
├── bin/
├── config/
│    └── default/
│         ├── config.json <1>
│         ├── dev-validators.json <2>
│         ├── genesis_assets.json <3>
│         ├── genesis_block.blob <4>
│         └── passphrase.json <5>
├── src/
└── test/
----

<1> The default configuration of the sidechain client exists in the `config.json` file.
<2> `dev-validators.json`: Contains the plain & encrypted keys, addresses, and other properties of 102 validators.
<3> `genesis_assets.json`: Contains details about the initial state of a sidechain such as account details for `token` and `pos` modules. It also contains the list of validators, etc.
<4> `genesis_block.blob`: A blob file of the genesis block.
<5> `passphrase.json`: Contains the passphrase used in various sidechain client operations.

== Why create a new genesis block?
For every new sidechain, a unique genesis block is required. 
As mentioned earlier, a default genesis block becomes available right after the client's `init` process. 
However, while running a public network, the developers have to create a new genesis block to change the initial state.

> The initial genesis block is expected to be used only for the development purposes.

Author: ```suggestion
The initial genesis block is expected to be used only for development purposes.
```

====

Each blockchain application needs a xref:{url_genesis_block}[genesis block], which is the very first block of the blockchain.

During the init process of a sidechain client, the lisk-sdk bootstraps the first genesis block required for each sidechain and the relevant assets in the `config/default` directory of each client.

.Default genesis block and configuration files
----
├── bin/
├── config/
│    └── default/
│         ├── config.json <1>
│         ├── dev-validators.json <2>
│         ├── genesis_assets.json <3>
│         ├── genesis_block.blob <4>
│         └── passphrase.json <5>
├── src/
└── test/
----

<1> The default configuration of the sidechain client exists in the `config.json` file.
<2> `dev-validators.json`: Contains the plain & encrypted keys, addresses, and other properties of 102 validators.
<3> `genesis_assets.json`: Contains details about the initial state of a sidechain such as account details for `token` and `pos` modules. It also contains the list of validators, etc.
<4> `genesis_block.blob`: A blob file of the genesis block.
<5> `passphrase.json`: Contains the passphrase used in various sidechain client operations.

== Why create a new genesis block?

> For every new sidechain, a unique genesis block is required. As mentioned earlier, a default genesis block becomes available right after the client's `init` process. However, while running a public network, you may want to change the initial state of the sidechain.

Reviewer: for the public network, developers have to create a new genesis block. Initial genesis block are expected to be used only for development

> To do that, you can regenerate a genesis block for various use cases, such as assigning different tokens to specific addresses of your choice.  You can also recreate the genesis block based on the needs of modules, for example, setting up the initial constraints required for a staking module, etc.

Reviewer 2: ```suggestion
To achieve this you can regenerate a genesis block for various use cases, such as assigning different tokens to specific addresses of your choice. Furthermore, it is also possible to recreate the genesis block based on the module's requirements, for example, setting up the initial constraints required for a staking module, etc.
```
